lib/restbase.js

```diff
@@ -243,6 +243,10 @@ RESTBase.prototype._request = function(req) {
     var match = priv.router.route(childReq.uri);
     var methods = match && match.value && match.value.methods;
     var handler = methods && (methods[childReq.method] || methods.all);
+    if (childReq.method === 'head' && !handler) {
```

Comment:
But this would make the req obj not monomorphic, but, actually we can fight this problem too, so I do agree, lets do it, I'll prepare a new PR tomorrow